solr/solrj/src/java/org/apache/solr/client/solrj/ResponseParser.java

```diff
@@ -16,45 +16,31 @@
  */
 package org.apache.solr.client.solrj;
 
+import java.io.IOException;
 import java.io.InputStream;
-import java.io.Reader;
 import java.util.Collection;
-import java.util.Set;
 import org.apache.solr.common.util.NamedList;
 
 /**
+ * SolrJ Solr response parser.
+ *
  * @since solr 1.3
  */
 public abstract class ResponseParser {
-  public abstract String getWriterType(); // for example: wt=XML, JSON, etc
 
-  public abstract NamedList<Object> processResponse(InputStream body, String encoding);
+  /** The writer type placed onto the request as the {@code wt} param. */
+  public abstract String getWriterType(); // for example: wt=XML, JSON, etc
 
-  public abstract NamedList<Object> processResponse(Reader reader);
-
-  /**
-   * A well-behaved ResponseParser will return its content-type.
-   *
-   * @return the content-type this parser expects to parse
-   * @deprecated use {@link #getContentTypes()} instead
-   */
-  @Deprecated
-  public String getContentType() {
-    return null;
-  }
+  public abstract NamedList<Object> processResponse(InputStream body, String encoding)
+      throws IOException;
 
   /**
    * A well-behaved ResponseParser will return the content-types it supports.
    *
-   * @return the content-type values that this parser is capable of parsing.
+   * @return the content-type values that this parser is capable of parsing. Never null. Empty means
+   *     no enforcement.
```

Comment:
Somewhat similar to the previous comment, I don't kno wthat "enforcement" is a thing I would expect on response parsers around content-types...

Reply:
I could easily imagine a misalignment between the response from Solr and the parser (different format).  This acts as a sanity check instead of blindly parsing what we could rule out with a friendly error message.